### PR TITLE
Render new contents for async panel rendering

### PR DESCRIPTION
**Pull request by nwalters512**
Part of #10792. Uses the new components introduced in #10942, which removes the need to have a CSRF token in this context. Because of the work done in #10936, this will be backwards-compatible with clients that are out there in the wild.

### CODE REVIEW

File: apps/prairielearn/src/pages/studentInstanceQuestion/studentInstanceQuestion.ts
Comment (nwalters512) [inline apps/prairielearn/src/pages/studentInstanceQuestion/studentInstanceQuestion.ts:266]: These are in some sense the most important changes in this PR: in the next PR, we can teach clients how to fetch rendered panels from this endpoint.

File: apps/prairielearn/src/lib/externalGradingSocket.ts
Comment (jonatanschroeder) [inline apps/prairielearn/src/lib/externalGradingSocket.ts:95]: You can remove the csrf_token from the `ensureProps` call a few lines before this point. The socket may still be sending them, but you don't need to enforce it is there.

File: apps/prairielearn/src/lib/externalGradingSocket.ts
Comment (jonatanschroeder) [inline apps/prairielearn/src/lib/externalGradingSocket.ts:96]: This is a bit of a nit, but I was testing this and checking in dev tools what was being sent through the socket, and I noticed that this causes the question footer to be re-rendered even in cases where it doesn't need to be (e.g., instructor question preview). I believe this is existing behaviour, so I won't block on that, I just wanted to point that out. Besides, this will eventually be removed once #10792 is fully merged, since it that case `renderScorePanels` will be set by the router code, not by the socket code.

**Review by jonatanschroeder — APPROVED** [review]

**Pull request merged**
